src/main/java/oscar/essential/TaskList.java

```diff
@@ -0,0 +1,169 @@
+        try {
+            @SuppressWarnings("unchecked")
+            ArrayList<Task> tempList = (ArrayList<Task>) stream.readObject();
+            taskList = tempList;
+            return taskList;
+        } catch (IOException e) {
+            throw new OscarException("Sorry! There is an error loading the saved task list.\n");
+        } catch (ClassNotFoundException e) {
+            throw new OscarException("Sorry! Class cannot be found.\n");
+        }
+    }
+
+    /**
+     * Saves the current task list by serializing it.
+     *
+     * @param stream Object output stream of saved file.
+     * @throws OscarException Input or output error.
+     */
+    public void save(ObjectOutputStream stream) throws OscarException {
+        try {
+            stream.writeObject(taskList);
+        } catch (IOException e) {
+            throw new OscarException("Sorry! There is an issue with your input or output.\n");
+        }
+    }
+
+    /**
+     * Obtains the number of tasks in the task list.
+     *
+     * @return Count of tasks.
+     */
+    public int getSize() {
+        return taskList.size();
+    }
+
+    /**
+     * Lists stored tasks in chronological order of addition.
+     *
+     * @return Tasks in task list.
+     */
+    public String list() {
+        StringBuilder result = new StringBuilder("Here are the tasks in your list:\n");
+        for (int i = 1; i <= taskList.size(); i++) {
+            Task currentTask = taskList.get(i - 1);
+            result.append(i).append(".").append(currentTask).append("\n");
+        }
+        return result.append("\n").toString();
+    }
+
+    /**
+     * Displays the number of tasks stored in the task list.
+     *
+     * @return Message that informs users of number of tasks in task list.
+     */
+    public String listCount() {
+        int listSize = getSize();
+        if (listSize == 0) {
+            return "You have no tasks in the list. Add some now!\n";
+        } else if (listSize == 1) {
+            return "You have 1 task in the list.\n";
+        } else {
+            return "You now have " + listSize + " tasks in the list.\n";
+        }
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param index Task number.
+     * @return Description of task.
+     */
+    public String mark(int index) {
+        Task currentTask = taskList.get(index);
+        currentTask.markAsDone();
+        return currentTask.toString();
+    }
+
+    /**
+     * Marks a task as not done.
+     *
+     * @param index Task number.
+     * @return Description of task.
+     */
+    public String unmark(int index) {
+        Task currentTask = taskList.get(index);
+        currentTask.markAsNotDone();
+        return currentTask.toString();
+    }
+
+    /**
+     * Deletes a task.
+     *
+     * @param index Task number.
+     * @return Description of task.
+     */
+    public String delete(int index) {
```

Comment:
Minor suggestion but perhaps it would be clearer if the method name was deleteTask()?